kbn-dev-gcp.sh

```diff
@@ -0,0 +1,121 @@
+    terraform -chdir="${SCRIPT_DIR}/gcp" apply \
+      -var="gcp_name=${gcp_name}" \
+      -var="kibana_repo_url=${repo_url}" \
+      -var="kibana_repo_branch=${branch}" \
+      -auto-approve
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace select "${workspace_id}"
+    public_ip=$(terraform -chdir="${SCRIPT_DIR}/gcp" output -json | jq  -r '.public_ip.value')
+    kibana_url=$(terraform -chdir="${SCRIPT_DIR}/gcp" output -json | jq  -r '.kibana_url.value')
+    if [[ $kibana_url != 'null' ]];
+      then
+        sed -i "/^${workspace_id}/d " $deployments_file
+        echo "${workspace_id}, ${gcp_name}, ${repo_url}, ${branch}, ${kibana_url}" >> $deployments_file
+        log "Success deploying instance ${kibana_url}"
+        log "Checking for server to be available"
+        eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${public_ip} /tmp/check_server_online.sh"
+        log "Checking for UI to be available"
+        eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${public_ip} /tmp/check_ui_online.sh"
+        log "UI is available"
+        END=$(date +%s)
+        DIFF=$(echo "$END - $START" | bc)
+        log "Whole process took ${DIFF} seconds"
+      fi
+    ;;
+
+  status)
+    echo "Status of ${type} ${value}, ${repo_url}, ${branch}"
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace select "${workspace_id}"
+    terraform -chdir="${SCRIPT_DIR}/gcp" output
+    ;;
+
+  ssh)
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace select "${workspace_id}"
+    public_ip=$(terraform -chdir="${SCRIPT_DIR}/gcp" output -json | jq  -r '.public_ip.value')
+    eval "ssh -o StrictHostKeyChecking=no ubuntu@${public_ip}"
+    ;;
+
+  update)
+    log "Updating instance of ${type} ${value}, ${repo_url}, ${branch}"
+    START=$(date +%s)
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace select "${workspace_id}"
+    terraform_output=$(terraform -chdir="${SCRIPT_DIR}/gcp" output -json | jq  -r '.public_ip.value')
+    eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${terraform_output} bash /tmp/update.sh"
+    eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${terraform_output} bash /tmp/bootstrap.sh"
+    eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${terraform_output} bash /tmp/start.sh"
+    eval "ssh -q -o StrictHostKeyChecking=no ubuntu@${terraform_output} bash /tmp/check_server_online.sh"
+    END=$(date +%s)
+    DIFF=$(echo "$END - $START" | bc)
+    log "Success updating instance (Duration: ${DIFF}s) of ${type} ${value}, ${repo_url}, ${branch}"
+    ;;
+
+  destroy)
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace select "${workspace_id}"
+    log "Destroying instance of ${type} ${value}, ${repo_url}, ${branch}"
+    terraform -chdir="${SCRIPT_DIR}/gcp" destroy -auto-approve
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace select default
+    terraform -chdir="${SCRIPT_DIR}/gcp" workspace delete "${workspace_id}"
+    sed -i "/^${workspace_id}/d " $deployments_file
```

Comment:
on `destroy` I was getting
```
sed: 1: "/Users/Luke/path/to/script ...": extra characters at the end of l command
```

is that just me?

Reply:
it's not just you ... it's also me ... so sed works a bit differently on OSX, and when I make it work for OSX, it doesn't work on Linux. Recently build for cloud and linux, because I'm thinking of a tool people can use without having to install local dependencies like terraform. do you know a bash trick to make it work for both envs? thx!

Reply:
heard this should work for both envs: 

```
sed -e '…' "$file" >"$file.new"
mv -- "$file.new" "$file"
```

Reply:
changed that